packages/braintree-integration/src/BraintreeAchPaymentMethod.tsx

```diff
@@ -18,37 +16,66 @@ const BraintreeAchPaymentMethod: FunctionComponent<PaymentMethodProps> = ({
     onUnhandledError,
     paymentForm,
 }) => {
+    const [isInitialize, setIsInitialize] = useState(false);
+    const [currentMandateText, setCurrentMandateText] = useState<string>('');
+    const currentMandateTextRef = useRef('');
+
     useEffect(() => {
+        currentMandateTextRef.current = currentMandateText;
+    }, [currentMandateText]);
+
+    const getMandateText = useCallback(() => {
+        return currentMandateTextRef.current;
+    }, []);
+
+    const initializePayment = useCallback(async () => {
         const { gateway: gatewayId, id: methodId } = method;
 
-        const initialize = async () => {
+        try {
+            await checkoutService.initializePayment({
+                gatewayId,
+                methodId,
+                braintreeach: {
+                    getMandateText,
+                },
+            });
+        } catch (error) {
+            if (error instanceof Error) {
+                onUnhandledError(error);
+            }
+        }
+    }, [checkoutService, getMandateText, method, onUnhandledError]);
+
+    useEffect(() => {
+        if (!isInitialize) {
+            setIsInitialize(true);
+            void initializePayment();
+        }
+    }, [initializePayment, isInitialize]);
+
+    useEffect(() => {
+        const initializeBillingAddressFields = async () => {
```

Comment:
Should we reinitialise BillingAddressFields each time on state change?

Reply:
BillingAddressFields will only be initialized once during bean initialization